server/app/services/scheduler/filter/volume_instance.rb

```diff
@@ -0,0 +1,49 @@
+module Scheduler
+  module Filter
+    class VolumeInstance
+      include Logging
+
+      ##
+      # @param [GridService] service
+      # @param [Integer] instance_number
+      # @param [Array<HostNode>] nodes
+      # @raise [Scheduler::Error]
+      def for_service(service, instance_number, nodes)
+        return nodes if service.service_volumes.empty?
+
+        debug "filtering for service #{service.to_path}-#{instance_number}"
+
+        filtered_nodes = nodes.dup
+
+        service.service_volumes.each do |sv|
+          if sv.volume
+            case sv.volume.scope
+            when 'instance'
+              # Try to find a node which has the volume already for the instance
+              filtered_nodes = filtered_nodes.select { |node|
+                node.volume_instances.where(name: sv.volume.name_for_service(service, instance_number)).exists?
+              }
+            when 'stack'
+              # Stack scoped volumes can be created per node
```

Comment:
Kinda questioning this assumption, it depends on the driver. This prevents using `driver: local` volumes with multiple services by scheduling all of those onto the same node.

We can fix this post-merge.